samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManagerFactory.scala

```diff
@@ -91,8 +99,10 @@ class KafkaCheckpointManagerFactory extends CheckpointManagerFactory with Loggin
       socketTimeout,
       consumerConfig.socketReceiveBufferBytes,
       consumerConfig.fetchMessageMaxBytes,            // must be > buffer size
+      systemConsumerF,
```

Comment:
s/systemConsumerF/systemConsumerFn

Similar for producer, and admin

Reply:
Prefer not extracting these functions. +1 for Jagadish's suggestion for taking a SystemAdmin. If not, they seem simple/short enough to inline.
If you do decide to extract them, better to rename these to get/createSystemXYZ, and remove the {} and make it a one liner. E.g.:
val createSystemConsumer =  () => systemFactory.getSystemConsumer(...)

Reply:
ok.